### DISCUSSION TITLE
Error: Detected Inf/NaN in energies. Aborting.

### DISCUSSION THREAD

**Discussion by Rahim-Habibi** [SeisSol/SeisSol#1138] [Q&A] [answered]
Hi all, 

i have got the following error message and then the run was terminated. I had some tries but the same error shows up at different time step and terminats the run. for examle once at 2.5 s, aonther time at 3.8s etc. 
So, what is the problem and how can i resolve it. here is the log in the end:

Sat Jul 27 13:33:04, Info:  Time spent this phase (IO): 153.1336 ms (min: 152.6048 ms, max: 153.6624 ms)
Sat Jul 27 13:33:04, Info:  Performance since the start: 975.6944 GFLOP/s (rank 0: 470.0271 GFLOP/s , average over ranks: 487.8472 GFLOP/s )
Sat Jul 27 13:33:04, Info:  Performance since last sync point: 975.4167 GFLOP/s (rank 0: 469.8933 GFLOP/s , average over ranks: 487.7083 GFLOP/s )
Sat Jul 27 13:33:20, Info:  #max-updates since sync:  6400  @  2.48753
Sat Jul 27 13:33:25, Info:  Writing faultoutput at time 2.5.
Sat Jul 27 13:33:25, Info:  Writing faultoutput at time 2.5. Done.
Sat Jul 27 13:33:25, Info:  Writing free surface at time 2.5.
Sat Jul 27 13:33:25, Info:  Writing free surface at time 2.5. Done.
Sat Jul 27 13:33:25, Info:  Writing energy output at time 2.5
Sat Jul 27 13:33:25, Info:  Elastic energy (total, % kinematic, % potential):  8.03797e+11  , 11.5371  , 88.4629
Sat Jul 27 13:33:25, Info:  Total momentum (X, Y, Z): 4.54597e+08  , -3.60373e+08  , 3.40525e+08
Sat Jul 27 13:33:25, Info:  Frictional work (total, % static, % radiated):  1.15232e+12  , 27.4382  , 72.5618
Sat Jul 27 13:33:25, Info:  Seismic moment (without plasticity): 5.46718e+15  Mw: 4.42184
Sat Jul 27 13:33:25, Info:  Writing energy output at time 2.5 Done.
Sat Jul 27 13:33:25, Info:  Time spent this phase (total): 20.8323 s (min: 20.8323 s, max: 20.8323 s)
Sat Jul 27 13:33:25, Info:  Time spent this phase (compute): 20.8323 s (min: 20.8322 s, max: 20.8323 s)
Sat Jul 27 13:33:25, Info:  Time spent this phase (IO): 193.1633 ms (min: 171.6557 ms, max: 214.6709 ms)
Sat Jul 27 13:33:25, Info:  Performance since the start: 975.7446 GFLOP/s (rank 0: 470.0513 GFLOP/s , average over ranks: 487.8723 GFLOP/s )
Sat Jul 27 13:33:25, Info:  Performance since last sync point: 978.2104 GFLOP/s (rank 0: 471.2393 GFLOP/s , average over ranks: 489.1052 GFLOP/s )
Sat Jul 27 13:33:41, Info:  #max-updates since sync:  6400  @  2.53753
Sat Jul 27 13:33:46, Info:  Writing energy output at time 2.55
Sat Jul 27 13:33:47, Info:  Elastic energy (total, % kinematic, % potential):  -nan  , -nan  , -nan
Sat Jul 27 13:33:47, Info:  Total momentum (X, Y, Z): -nan  , -nan  , -nan
Sat Jul 27 13:33:47, Info:  Frictional work (total, % static, % radiated):  -nan  , -nan  , -nan
Sat Jul 27 13:33:47, Info:  Seismic moment (without plasticity): -nan  Mw: -nan
Sat Jul 27 13:33:47, Error: Detected Inf/NaN in energies. Aborting.
Backtrace:
SeisSol_Release_srome_4_elastic(_ZN5utils6LoggerD1Ev+0x128) [0x561eed4bee08]
SeisSol_Release_srome_4_elastic(+0xc85b1) [0x561eed5095b1]
SeisSol_Release_srome_4_elastic(+0xca2d8) [0x561eed50b2d8]
SeisSol_Release_srome_4_elastic(+0x2177aa) [0x561eed6587aa]
SeisSol_Release_srome_4_elastic(+0x217d57) [0x561eed658d57]
SeisSol_Release_srome_4_elastic(+0x235048) [0x561eed676048]
SeisSol_Release_srome_4_elastic(+0xf7842) [0x561eed538842]
SeisSol_Release_srome_4_elastic(+0x4c8a4) [0x561eed48d8a4]
/lib/x86_64-linux-gnu/libc.so.6(+0x2724a) [0x7faa5658124a]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0x85) [0x7faa56581305]
SeisSol_Release_srome_4_elastic(+0x6d611) [0x561eed4ae611]
[heisenbug:3517486] *** Process received signal ***
[heisenbug:3517486] Signal: Aborted (6)
[heisenbug:3517486] Associated errno: Unknown error 32682 (32682)
[heisenbug:3517486] Signal code:  (27)
[heisenbug:3517486] [ 0] /lib/x86_64-linux-gnu/libc.so.6(+0x3c050)[0x7faa56596050]
[heisenbug:3517486] [ 1] /lib/x86_64-linux-gnu/libc.so.6(+0x8ae2c)[0x7faa565e4e2c]
[heisenbug:3517486] [ 2] /lib/x86_64-linux-gnu/libc.so.6(gsignal+0x12)[0x7faa56595fb2]
[heisenbug:3517486] [ 3] /lib/x86_64-linux-gnu/libc.so.6(abort+0xd3)[0x7faa56580472]
[heisenbug:3517486] [ 4] SeisSol_Release_srome_4_elastic(+0x7dec0)[0x561eed4beec0]
[heisenbug:3517486] [ 5] SeisSol_Release_srome_4_elastic(+0xc85b1)[0x561eed5095b1]
[heisenbug:3517486] [ 6] SeisSol_Release_srome_4_elastic(+0xca2d8)[0x561eed50b2d8]
[heisenbug:3517486] [ 7] SeisSol_Release_srome_4_elastic(+0x2177aa)[0x561eed6587aa]
[heisenbug:3517486] [ 8] SeisSol_Release_srome_4_elastic(+0x217d57)[0x561eed658d57]
[heisenbug:3517486] [ 9] SeisSol_Release_srome_4_elastic(+0x235048)[0x561eed676048]
[heisenbug:3517486] [10] SeisSol_Release_srome_4_elastic(+0xf7842)[0x561eed538842]
[heisenbug:3517486] [11] SeisSol_Release_srome_4_elastic(+0x4c8a4)[0x561eed48d8a4]
[heisenbug:3517486] [12] /lib/x86_64-linux-gnu/libc.so.6(+0x2724a)[0x7faa5658124a]
[heisenbug:3517486] [13] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0x85)[0x7faa56581305]
[heisenbug:3517486] [14] SeisSol_Release_srome_4_elastic(+0x6d611)[0x561eed4ae611]
[heisenbug:3517486] *** End of error message ***
--------------------------------------------------------------------------
prterun noticed that process rank 0 with PID 3517486 on node heisenbug exited on
signal 6 (Aborted).

Thanks in advance,
Rahim

**Comment by davschneller** [comment]
Hi Rahim,

an `Inf/NaN` error usually indicates that the simulation somehow diverged; usually that's related to the model you want to simulate—some are just a bit more "picky" about when they work and not. As solution: you could try using `double` precision for the simulation. If that still doesn't do it, you can use a higher order. If nothing of that worked—we may need to take a closer look at the model. :)

Cheers,
David

**Reply by Rahim-Habibi** [reply, marked as answer]
Thanks @davschneller 

I have resolved the issue. Actually in my case, it was meshing problem which causing divergence after a certain time step.
Rahim
